src/MarketingPageAcceptanceTestsSpecflow/Steps/ClientApplication/HardwareRequirements/HardwareRequirements.cs

```diff
@@ -10,46 +10,62 @@ public HardwareRequirements(UITest test, ScenarioContext context) : base(test, c
         {
         }
 
-        [Given(@"the Supplier has entered text on the (.*) page")]
-        public void GivenTheSupplierHasEnteredText(string page)
+        [Given(@"the Supplier has entered (\d{3,4}) characters on the (.*) page")]
+        public void GivenTheSupplierHasEnteredText(int characters, string page)
         {
-            _context.Pending();
+            _test.pages.Dashboard.NavigateToSection("Client application type");
+            _test.pages.ClientApplicationTypes.SelectCheckbox("Browser based");
+            _test.pages.ClientApplicationTypes.SaveAndReturn();
+            _test.pages.Dashboard.NavigateToSection("Browser based", true);
+
+            _test.pages.BrowserSubDashboard.OpenSection(page);
+
+            _test.pages.HardwareRequirements.EnterText(characters);
         }
 
-        [Given(@"the Browser Based Client Application Type Sub-Section does not require Mandatory Data")]
-        public void GivenTheBrowserBasedClientApplicationTypeSub_SectionDoesNotRequireMandatoryData()
+        [Given(@"the (.*) Sub-Section does not require Mandatory Data")]
+        public void GivenTheSub_SectionDoesNotRequireMandatoryData(string sectionName)
         {
-            _context.Pending();
+            _test.pages.Dashboard.NavigateToSection("Client application type");
+            _test.pages.ClientApplicationTypes.SelectCheckbox("Browser based");
+            _test.pages.ClientApplicationTypes.SaveAndReturn();
+            _test.pages.Dashboard.NavigateToSection("Browser based", true);
```

Comment:
shouldn't you be navigating to sectionName?

Reply:
no, that bit is navigating to the browser based dashboard in preparation for navigating to the section